src/compiler/checker.ts

```diff
@@ -14301,11 +14301,11 @@ namespace ts {
                         candidateName === "set") {
                         continue;
                     }
-                    const distance = levenshtein(name, candidateName);
+                    const distance = dameraulevenshtein(name, candidateName);
                     if (distance > worstDistance) {
                         continue;
                     }
-                    if (distance < 3) {
+                    if (distance <= MIN_LEV_DIST) {

```

Comment:
That would make my example above revert to the prior behavior where we pick `twnty` over `twenty` because it's "good enough", when it's really not.